class.jetpack.php

```diff
@@ -926,33 +927,39 @@ function point_edit_post_links_to_calypso( $default_url, $post_id ) {
 		// Mapping the allowed CPTs on WordPress.com to corresponding paths in Calypso.
 		// https://en.support.wordpress.com/custom-post-types/
 		$allowed_post_types = array(
-			'post'                => 'post',
-			'page'                => 'page',
-			'jetpack-portfolio'   => 'edit/jetpack-portfolio',
-			'jetpack-testimonial' => 'edit/jetpack-testimonial',
+			'post',
+			'page',
+			'jetpack-portfolio',
+			'jetpack-testimonial',
 		);
 
-		if ( ! in_array( $post_type, array_keys( $allowed_post_types ) ) ) {
+		if ( ! in_array( $post_type, $allowed_post_types, true ) ) {
 			return $default_url;
 		}
 
-		$path_prefix = $allowed_post_types[ $post_type ];
-
-		$site_slug = self::build_raw_urls( get_home_url() );
-
-		return esc_url( sprintf( 'https://wordpress.com/%s/%s/%d', $path_prefix, $site_slug, $post_id ) );
+		return esc_url(
+			Redirect::get_url(
+				'calypso-edit-' . $post_type,
+				array(
+					'path' => $post_id,
+				)
+			)
+		);
 	}
 
 	function point_edit_comment_links_to_calypso( $url ) {
 		// Take the `query` key value from the URL, and parse its parts to the $query_args. `amp;c` matches the comment ID.
 		wp_parse_str( wp_parse_url( $url, PHP_URL_QUERY ), $query_args );
+
 		return esc_url(
-			sprintf(
-				'https://wordpress.com/comment/%s/%d',
-				self::build_raw_urls( get_home_url() ),
-				$query_args['amp;c']
+			Redirect::get_url(
+				'calypso-edit-comment',
```

Comment:
I think it's hard to keep full consistency on these slugs. 

Here it is matching the patter to all edit urls such as `calypso-edit-[post|page|pages|jetpack-portfolio|...]`

If we kept as it is in the url, we would not be able to simply concatenate different post types in the slugs, as we saw in the comment above.

It's a trade-off. What I have in mind is that on will have to always look up in the list of redirects anyway to add a new link, it will be hard to deduce the slug based on the URL.